cmd/commands/server/command_server.go

```diff
@@ -158,14 +167,90 @@ func (cmd *Command) Start() (err error) {
 		return err
 	}
 
-	signer := cmd.createSigner(providerID)
+	providerAddress := common.HexToAddress(providerID.Address)
+	// check if node's identity is registered
+	registered, err := identityRegistry.IsRegistered(providerAddress)
+	if err != nil {
+		return err
+	}
 
+	registrationDataProvider := registry.NewRegistrationDataProvider(cmd.keystore)
+	signer := cmd.createSigner(providerID)
 	cmd.proposalAnnouncementStopped.Add(1)
-	go cmd.discoveryAnnouncementLoop(proposal, cmd.mysteriumClient, signer, stopDiscoveryAnnouncement)
+
+	// if not registered - wait indefinitely for identity registration event
+	if !registered {
```

Comment:
This looks very complicated and could possibly refactored to separate function loop as:
1. check registration
2. wait if needed
3. register proposal
4. ping proposal
5. unregister proposal
It could be discovery component of its own